src/hutch_bunny/core/AvailabilityQuerySolver.py

```diff
@@ -0,0 +1,406 @@
+            ALL GROUPS COMPLETED, NOW APPLY LOGIC BETWEEN GROUPS
+            """
+
+            # construct the query based on the OR/AND logic specified between groups
+            if self.query.cohort.groups_operator == "OR":
+                full_query_all_groups = select(func.count()).where(or_(*all_groups_queries))
+            else:
+                full_query_all_groups = select(func.count()).where(*all_groups_queries)
+
+            # here for debug, prints the SQL statement created
+            print(str(full_query_all_groups.compile(
+                dialect=postgresql.dialect(),
+                compile_kwargs={"literal_binds": True})))
+
+            output = con.execute(full_query_all_groups).fetchone()
+
+        return int(output[0])
+    def addRelativeTime(self, left_value_time, right_value_time):
+        relative_date = self.calc_relative_date(left_value_time, right_value_time)
+
+        # if the left value is blank, it means the original was |1 meaning
+        # "I want to find this event that occurred less than a month ago"
+        # therefore the logic is to search for a date that is after the date
+        # that was a month ago.
+        if left_value_time == "":
+            self.meas = self.meas.where(Measurement.measurement_date >= relative_date)
+            self.obs = self.obs.where(Observation.observation_date >= relative_date)
+            self.condition = self.condition.where(ConditionOccurrence.condition_start_date >= relative_date)
+            self.drug = self.drug.where(DrugExposure.drug_exposure_start_date >= relative_date)
+        else:
+            self.meas = self.meas.where(Measurement.measurement_date <= relative_date)
+            self.obs = self.obs.where(Observation.observation_date <= relative_date)
+            self.condition = self.condition.where(ConditionOccurrence.condition_start_date <= relative_date)
+            self.drug = self.drug.where(DrugExposure.drug_exposure_start_date <= relative_date)
+    def addConceptSearch(self, current_rule: Rule):
+        if current_rule.operator == "=":
+            condition = self.condition.where(ConditionOccurrence.condition_concept_id == int(current_rule.value))
+            drug = self.drug.where(DrugExposure.drug_concept_id == int(current_rule.value))
+            meas = self.meas.where(Measurement.measurement_concept_id == int(current_rule.value))
+            obs = self.obs.where(Observation.observation_concept_id == int(current_rule.value))
+        else:
+            condition = self.condition.where(ConditionOccurrence.condition_concept_id != int(current_rule.value))
+            drug = self.drug.where(DrugExposure.drug_concept_id != int(current_rule.value))
+            meas = self.meas.where(Measurement.measurement_concept_id != int(current_rule.value))
+            obs = self.obs.where(Observation.observation_concept_id != int(current_rule.value))
+
+    def addRelativeAgeSearch(self, left_value_time: str, right_value_time: str):
```

Comment:
I'd suggest given `right_value_time` and `left_value_time` are cast to integers when they are used, this method should just take them as integers. 
Therefore it is up to wherever it is used to do the casting. It shouldn't be the responsibility of this method to do conversion.